### ISSUE TITLE
String parsing with new lines is broken

### ISSUE THREAD

**Issue by Necr0x0Der** [trueagi-io/hyperon-experimental#780]
**Describe the bug**
After this PR:
https://github.com/trueagi-io/hyperon-experimental/pull/777/
unit tests in metta-motto became broken.

**To Reproduce**
Run the test in metta-motto:
```
!(assertEqual
   ((echo-agent)
    (system "Ping") (user "Pong"))
"system Ping
user Pong")
```
While `echo-agent` returns
```
"system Ping
user Pong"
```
which is composed as `'\n'.join("system Ping", "user Pong")`, it cannot be matched with the string formed by parsing in Metta
```
"system Ping
user Pong"
```
This:
```
!(assertEqual
   ((echo-agent)
    (system "Ping") (user "Pong"))
"system Ping\nuser Pong")
```
also doesn't work.

**Expected behavior**
There should be a way to specify multi-line strings in metta scripts, which could be matched against corresponding strings formed in Python.

**Comment by Necr0x0Der** [comment]
The issue seems that
```
"system Ping
user Pong"
```
is not converted into Python String by
```
r"^\".*\"$": lambda token: ValueAtom(str(token[1:-1]), 'String'),
```
and is parsed as Rust string.
An attempt to write:
```
m = MeTTa()
x = m.run('''
! "A
B"
''')
print(x[0][0].get_object().content == '''A
B''')
```
results in
```
Cannot get_object of unsupported non-C "A
B"
```
Basically, the corresponding regex in Python is not invoked in this case. While if we return the old one
```
"\"[^\"]*\"": lambda token: ValueAtom(str(token[1:-1]), 'String')
```
it works here.

**Comment by vsbogd** [comment]
Interesting, probably, because `$` matches end of line.

**Comment by vsbogd** [comment]
Have you tried using `\n` in MeTTa as well?
```
"system Ping\nuser Pong"
```

**Comment by vsbogd** [comment]
I tried running the following script:
```
!(get-metatype "abc\ndef")
!(get-metatype "abc
def")
!(get-type "abc\ndef")
!(get-type "abc
def")
!(println! "abc\ndef")
!(println! "abc
def")
```
In both Python and Rust it shows the same result:
```
$ metta /tmp/test.metta
abc
def
abc
def
[Grounded]
[Grounded]
[String]
[String]
[()]
[()]
$ cargo run /tmp/test.metta
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.04s
     Running `/home/vital/projects/trueagi/hyperon-experimental/target/debug/metta-repl /tmp/test.metta`
abc
def
abc
def
[Grounded]
[Grounded]
[String]
[String]
[()]
[()]
```

**Comment by vsbogd** [comment]
First guess was correct: `.` inside regex doesn't recognize `\n` and because of this `te\nst` is parsed as a symbol atom. Fixed in #783
